### PR TITLE
feat(ME-3477): add private network flag and messages

### PR DESCRIPTION
**Pull request by btoonk**
## Summary by CodeRabbit

- **New Features**
	- Introduced new message types for peer-to-peer communication, including DiscoveryDetailsMessage, HeartbeatMessage, PeerOnlineMessage, and others.
	- Added a field to indicate private network functionality in the configuration settings.

- **Bug Fixes**
	- Improved Docker container access by updating volume mounts for broader project file access.
	- Corrected the return type of the ControlStream RPC method for better functionality.

- **Chores**
	- Updated the Protocol Buffers compiler command to reflect changes in directory structure for protocol files.

### CODE REVIEW

File: connector/connector.proto [hunk 1]
```diff
@@ -28,6 +29,7 @@ message ControlStreamRequest {
     UploadRecording upload_recording = 11;
     CertifcateSignRequest certifcate_sign_request = 12;
     SessionEvent session_event = 13;
+    DiscoveryDetailsMessage discoveryDetailsMessage = 14;
```
[the remark this answers is not in the document]
Comment (adrianosela) [reply connector/connector.proto:32]: This is a good comment.

File: connector/connector.proto [hunk 2]
```diff
@@ -44,6 +46,10 @@ message ControlStreamReponse {
     AuthorizeResponse authorize = 9;
     SshCertifcateSignResponse ssh_certificate_sign_response = 10;
     CertifcateSignResponse certifcate_sign_response = 11;
+    border0.device.v1.NetworkStateMessage network_state_message = 12;
+    border0.device.v1.PeerOnlineMessage peerOnlineMessage = 13;
+    border0.device.v1.PeerOfflineMessage peerOfflineMessage = 14;
+    border0.device.v1.DisconnectMessage disconnectMessage = 15;
```
[the remark this answers is not in the document]
Comment (adrianosela) [reply connector/connector.proto:52]: good comment.

**Review by adrianosela — COMMENTED** [review]
I think we should make a new package for the device types and import it in both the connector and device packages. Otherwise its too easy to forget that connector imports it and accidentally make breaking changes? Maybe not since you would see the diff -- but i think its better.

File: Makefile [hunk 3]
```diff
@@ -7,12 +7,12 @@ PROTOC_GEN_GO_VERSION:=v1.34.2
 PROTOC_GEN_GO_GRPC_VERSION:=v1.5.1
 
 # add any components here after their directory has been created with .proto files
-COMPONENTS:=connector device 
+COMPONENTS:=common connector device
```
Comment (adrianosela) [inline Makefile:10]: don't need common here since its being mounted as a separate volume below.

File: common/messages.proto [hunk 4]
```diff
@@ -0,0 +1,55 @@
+// sets the .proto file syntax version
```
Comment (adrianosela) [inline common/messages.proto:1]: device.proto more accurate?

**Review by adrianosela — APPROVED** [review]
LGTM, some nit comments left.

File: Makefile (hunk 3 above)
Comment (btoonk) [reply Makefile:10]: but we still need to built it to get the go code.

File: common/messages.proto (hunk 4 above)
Comment (btoonk) [reply common/messages.proto:1]: it's not just devices (eg. there is also healtcheck).
otherwise we should split it up. For now I will keep it like this.

**Pull request merged**